samtranslator/model/stepfunctions/generators.py

```diff
@@ -129,20 +132,17 @@ def to_cloudformation(self):  # type: ignore[no-untyped-def]
             )
 
         if self.role and self.policies:
-            raise InvalidResourceException(
-                self.logical_id, "Specify either 'Role' or 'Policies' property and not both."
-            )
+            raise InvalidResourceException(self.logical_id, self.SFN_INVALID_PROPERTY_BOTH_ROLE_POLICY)
         if self.role:
             self.state_machine.RoleArn = self.role
-        elif self.policies:
-            if not self.managed_policy_map:
+        else:
+            if self.policies and not self.managed_policy_map:
                 raise Exception("Managed policy map is empty, but should not be.")
-
+            if not self.policies:
+                self.policies = []
```

Comment:
Is it it? Where do we actually create a role or a policy?

Reply:
in `_construct_role()` in generator.py

Reply:
Yeah, in _constructor_role() a role is generator based on a policy.